tetris.js

```diff
@@ -317,6 +303,44 @@ module.exports = {
 		for(let arrayElement of arrayCoordinates){
 			localArrayCoordinates.push(this.rotateElement(arrayElement,coordinatePivot[0]))	
 		}
-		return localArrayCoordinates
+		return this.sort(localArrayCoordinates)
+	},
+	sort: function(arrayOfCoordinates){
```

Comment:
1. Your imlementation is very complex and probably has a lot of bugs
2. Javascript already has array sort method. Did you google before writing your own? This method should be single line of code

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Syntax